File: php/Memory.php
<?php

class Memory extends Hardware
{
    private $frequency = "";
    private $memorySize = "";
    private $supportsCuda = "";

    public function __construct($f="", $ms="", $sc="")
    {
        $this->frequency = $f;
        $this->memorySize = $ms;
        $this->supportsCuda = $sc;
    }

    public function setFrequency($f = "")
    {
        $this->frequency = $f;
    }

    public function getFrequency()
    {
        return $this->frequency;
    }

    public function setMemorySize($ms = "")
    {
        $this->memorySize = $ms;
    }

    public function getMemorySize()
    {
        return $this->memorySize;
    }

    public function setSupportsCuda($sc = "")
    {
        $this->supportsCuda = $sc;
    }

    public function getSupportsCuda()
    {
        return $this->supportsCuda;
    }
}


?>

File: c++/Memory.cpp
#include "Hardware.cpp"

class Memory : public Hardware
{
    private:
        /* atribut private */
        string frequency;
        string memorySize;
        string supportsCuda;
    public:
        Memory(){
        }

        Memory(string f, string ms, string sc){
            frequency = f;
            memorySize = ms;
            supportsCuda = sc;
        }

        //membuat getter dan setter
        void setFrequency(string f){
            frequency = f;
        }

        string getFrequecy(){
            return frequency;
        }

        void setMemorySize(string ms){
            memorySize = ms;
        }

        string getMemorySize(){
            return memorySize;
        }

        void setSupportsCuda(string sc){
            supportsCuda = sc;
        }

        string getSupportsCuda(){
            return supportsCuda;
        }

        //prosedur menampilkan isi kelas
        void showMemory(){
            cout << "Memory Size   : " << memorySize << endl;
            cout << "Frequency     : " << frequency << endl;
            cout << "Supports Cuda : " << supportsCuda << endl;
        }

        ~Memory();
};